pkg/kubectl/cmd/scale.go

```diff
@@ -110,13 +110,24 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 	}
 
 	resourceVersion := cmdutil.GetFlagString(cmd, "resource-version")
+	if len(resourceVersion) != 0 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --resource-version with multiple controllers")
+	}
 	currentSize := cmdutil.GetFlagInt(cmd, "current-replicas")
+	if currentSize != -1 && len(infos) > 1 {
+		return fmt.Errorf("cannot use --current-replicas with multiple controllers")
+	}
 	precondition := &kubectl.ScalePrecondition{currentSize, resourceVersion}
 	retry := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
-	waitForReplicas := kubectl.NewRetryParams(kubectl.Interval, kubectl.Timeout)
```

Comment:
kubectl e2e will probably start flaking (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/kubectl.go#L91). A better way to do this would be through a timeout flag, like we do for --delete (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/delete.go#L80). You can default it to 0 meaning don't wait (and update the test so it stays at the default).